src/core/lib/iomgr/work_serializer.cc

```diff
@@ -35,16 +35,17 @@ struct CallbackWrapper {
   const DebugLocation location;
 };
 
-// First 32 bits indicate ownership of the WorkSerializer, next 32 bits are
+// First 16 bits indicate ownership of the WorkSerializer, next 48 bits are
 // queue size (i.e., refs).
-static uint64_t MakeRefPair(uint32_t owners, uint32_t size) {
+uint64_t MakeRefPair(uint16_t owners, uint64_t size) {
+  GPR_ASSERT(size >> 48 == 0);
   return (static_cast<uint64_t>(owners) << 32) + static_cast<int64_t>(size);
```

Comment:
Doesn't this now need to shift by 48 instead of 32?  How is the test passing with this bug?

Reply:
They are failing as expected. Fixed :)